### PR TITLE
[OV] Sets symbols on inputs and variables

### PR DESCRIPTION
**Pull request by jane-intel**
# What does this PR do?

OpenVINO exporter sets symbols for model inputs. It helps symbolic propagation and optimizations in OpenVINO.

OpenVINO provides symbolic shape inference and optimizations. By default, they deduce dimension relations from the model structure. However, in some cases it is impossible to deduce dimension relations, therefore user-defined symbols allow for smoother symbol propagation.

> Note: this change would only be working starting from 2024.3 OpenVINO release as it relies on new features of OpenVINO PyTorch FrontEnd

## Before submitting
- [N/A] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case). // No
- [N/A] Did you make sure to update the documentation with your changes? // No impact on documentation
- [N/A] Did you write any new necessary tests? // No, the task is complete

### CODE REVIEW

**Comment by slyalin** [comment]
@jane-intel, please temporarily update the dependency to the latest OV release candidate, then @echarlaix can unlock pre-commit tests.

**Comment by jane-intel** [comment]
@slyalin 24.3 release has already been published, so there is no need to change anything.
@echarlaix please unlock the pre-commit

**Comment by jane-intel** [comment]
There is a lower bound which is bumped due to backward incompatible change in optimum:
old PyTorch FE couldn't set shapes for tuple/dict inputs.

**Comment by jane-intel** [comment]
@echarlaix I applied style changes, could you please approve this PR again?

File: optimum/exporters/openvino/model_patcher.py [hunk 1]
```diff
@@ -1949,9 +1949,9 @@ def _dbrx_update_causal_mask_legacy(
                 offset = 0
             mask_shape = attention_mask.shape
             mask_slice = (attention_mask.eq(0.0)).to(dtype=dtype) * min_dtype
-            causal_mask[
-                : mask_shape[0], : mask_shape[1], offset : mask_shape[2] + offset, : mask_shape[3]
-            ] = mask_slice
+            causal_mask[: mask_shape[0], : mask_shape[1], offset : mask_shape[2] + offset, : mask_shape[3]] = (
+                mask_slice
+            )
```
Comment (eaidova) [inline optimum/exporters/openvino/model_patcher.py:1954]:
```suggestion
            causal_mask[
                : mask_shape[0], : mask_shape[1], offset : mask_shape[2] + offset, : mask_shape[3]
            ] = mask_slice
```

**Review by eaidova — APPROVED** [review]

**Review by IlyasMoutawwakil — APPROVED** [review]
LGTM! 
@jane-intel can you please run styling

**Comment by jane-intel** [comment]
Thank you @IlyasMoutawwakil, @echarlaix, and @eaidova for keeping PR up to date. I've applied style, @echarlaix could you approve to start the pre-commit?

**Comment by jane-intel** [comment]
@echarlaix thank you for the commit! However, I see that style check doesn't go through.

File: optimum/exporters/openvino/convert.py [hunk 2]
```diff
@@ -374,13 +371,12 @@ def ts_patched_forward(*args, **kwargs):
...
-                input_names = list(inputs.keys())
-                output_names = list(config.outputs.keys())
-                input_info = get_input_shapes(dummy_inputs, inputs)
-
-                ov_model = convert_model(model, example_input=dummy_inputs, input=input_info)
+                input_info = get_input_info(model, config, dummy_inputs)
+                ov_model = convert_model(
+                    model,
+                    example_input=dummy_inputs,
+                    input=list(map(lambda item: (item.shape, item.type), input_info)),
```
Comment (echarlaix) [inline optimum/exporters/openvino/convert.py:378]:
```suggestion
                    input=[(item.shape, item.type) for item in input_info],
```

will also fix the quality test https://github.com/huggingface/optimum-intel/actions/runs/10789831892/job/29923669303?pr=696

```
optimum/exporters/openvino/convert.py:378:27: C417 Unnecessary `map` usage (rewrite using a `list` comprehension)
optimum/exporters/openvino/convert.py:417:23: C417 Unnecessary `map` usage (rewrite using a `list` comprehension)
optimum/exporters/openvino/utils.py:89:18: C408 Unnecessary `list` call (rewrite as a literal)
optimum/exporters/openvino/utils.py:91:22: C408 Unnecessary `dict` call (rewrite as a literal)
```

File: optimum/exporters/openvino/convert.py [hunk 3]
```diff
@@ -411,27 +407,17 @@ def ts_patched_forward(*args, **kwargs):
...
-        flatten_inputs = flattenize_inputs(ordered_dummy_inputs.values())
-        ov_model.validate_nodes_and_infer_types()
+        ov_model.validate_nodes_and_infer_types()  # TODO: remove as unnecessary validation?
+
+        output_names = list(config.outputs.keys())
         for idx, out_tensor in enumerate(ov_model.outputs):
             if idx < len(output_names):
                 out_tensor.get_tensor().set_names({output_names[idx]})
 
+        input_names = list(map(lambda item: item.name, input_info))
```
Comment (echarlaix) [inline optimum/exporters/openvino/convert.py:417]:
```suggestion
        input_names = [item.name for item in input_info]
```

**Pull request merged**